src/main/java/edu/gatech/cs2340/trydent/Keyboard.java

```diff
@@ -0,0 +1,93 @@
+package edu.gatech.cs2340.trydent;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javafx.scene.input.KeyCode;
+/**
+ * A static keyboard class for ease of using key events.
+ *
+ */
+public class Keyboard {
+    private static Set<KeyCode> down = new HashSet<>();
+    private static Set<KeyCode> released = new HashSet<>();
+    private static Set<KeyCode> pressed = new HashSet<>();
+    private static Set<KeyCode> lastDownBuffer = new HashSet<>();
+    private static Set<KeyCode> downBuffer = new HashSet<>();
+    private static Set<KeyCode> releasedBuffer = new HashSet<>();
+    private static Set<KeyCode> pressedBuffer = new HashSet<>();
+
+    private static final Object REPLACE_LOCK = new Object();
+    private static final Object UPDATE_LOCK = new Object();
+
+    /**
+     * Determines if a key is down on the keyboard.
+     * @param key the key to check
+     * @return whether the key is currently down
+     */
+    public static boolean isKeyDown(KeyCode key) {
+        synchronized(REPLACE_LOCK) {
+            return down.contains(key);
+        }
+    }
+
+    /**
+     * Determines if a key is pressed just now.
+     * @param key the key to check
+     * @return whether the key is down starting with this frame
+     */
+    public static boolean isKeyDownOnce(KeyCode key) {
+        synchronized(REPLACE_LOCK) {
+            return pressed.contains(key);
+        }
+    }
+
+    /**
+     * Determines if a key is released just now.
+     * @param key the key to check
+     * @return whether the key is is up starting with this frame
+     */
+    public static boolean isKeyUpOnce(KeyCode key) {
+        synchronized(REPLACE_LOCK) {
+            return released.contains(key);
+        }
+    }
+
+    static void newFrame(){
+        synchronized(UPDATE_LOCK){
+            synchronized(REPLACE_LOCK) {
```

Comment:
In all seriousness, REPLACE_LOCK is unnecessary because the way the code is run,
newFrame and the public accessor methods are serialized.
But in the case there are three threads each running each method, deadlocks won't happen because the single lock methods do not depend on each other which means circular dependency is not possible.

But I guess I'll remove it because it is slows stuff down.

Refer to the necessary conditions for deadlocks: http://en.wikipedia.org/wiki/Deadlock